api/ifc/ifc.go

```diff
@@ -28,12 +28,23 @@ type KvLoader interface {
 
 // Loader interface exposes methods to read bytes.
 type Loader interface {
+
+	// IsRemote returns whether this Loader was created from a url
+	IsRemote() bool
+
+	// Repo returns the repo location of this Loader if IsRemote is true,
+	// and an empty string otherwise
+	Repo() string
```

Comment:
It would be helpful if you could explain why this is needed and where would you use this. In general, there isn't a big issue with adding methods to our interfaces, but this seems like it could potentially be a field in a struct or a local variable that is passed around. With some more context, we can make a better judgment about the recommendation. 

IIRC this change was a prerequisite for https://github.com/kubernetes-sigs/kustomize/pull/4652, so maybe it makes sense for you to include this change in that PR as that will provide the context and help reviewers to see how you are planning to use it.

Reply:
I will move the `Repo()` change into #4652 and keep auxiliary changes here.